src/resend.ts

```diff
@@ -40,7 +42,12 @@ export class Resend {
 
     if (!response.ok) {
       const error = await response.json();
-      return error;
+
+      if (isResendErrorResponse(error)) {
+        throw new ResendError(error.error.message, error.error.name, error.error.statusCode);
```

Comment:
I'm all for result types since they're more obvious. If doing so we should also export a type guard to help with inference 😃

Reply:
great idea! is that something you can work on?

Reply:
Of course, I'll update this tomorrow morning 😄

Reply:
great! I really appreciate it, we are preparing a major release for this package and having better error handling is going to be essential, thanks for looking into it 😄

Reply:
G'day sorry I had updated but said change is quite breaking since you're changing actual response types.

By having:

```
{
   data: ...
   error: undefined
}
```

We are changing most everything in the SDK, I'm happy to proceed with these updates but it might also be a nice idea to stop and consider if we can do:

```
type Response = Success | Error
```

Which is more achievable without introducing breaking changes.

Reply:
hey @Mythie, yeah, let's start with something more simple for now, I'm happy with either so doing `Success | Error` is a good step! I would just be wary of actually throwing errors, so as long we are returning them instead it's all good

Reply:
Sorry for the delay! Made said updates 😄